org.gnu.Octave.json

```diff
@@ -0,0 +1,186 @@
+{
+	"app-id": "org.gnu.Octave",
+	"runtime": "org.kde.Platform",
+	"runtime-version": "master",
+	"sdk": "org.kde.Sdk",
+	"command": "octave",
+	"rename-icon": "octave",
+	"rename-desktop-file": "www.octave.org-octave.desktop",
+	"rename-appdata-file": "www.octave.org-octave.appdata.xml",
+	"finish-args": [
+		"--socket=wayland",
+		"--socket=x11",
+		"--socket=pulseaudio",
+		"--share=network",
+		"--device=dri",
+		"--filesystem=home"
+	],
+	"build-options": {
+		"cflags": "-O2",
+		"cxxflags": "-O2"
+	},
+	"cleanup": [
+		"/include",
+		"/lib/pkgconfig",
+		"/lib/debug",
+		"/lib/cmake",
+		"*.a",
+		"*.la"
+	],
+	"modules": [
+		{
+			"name": "gmp",
+			"config-opts": [
+				"--enable-shared",
+				"--disable-static"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://gmplib.org/download/gmp/gmp-6.1.2.tar.xz",
+					"sha256": "87b565e89a9a684fe4ebeeddb8399dce2599f9c9049854ca8c0dfbdea0e21912"
+				}
+			]
+		},
+		{
+			"name": "mpfr",
+			"config-opts": [
+				"--enable-shared",
+				"--disable-static"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "http://www.mpfr.org/mpfr-current/mpfr-3.1.5.tar.gz",
+					"sha256": "f4eb5070883aee3fd8b927751ea63ff95aebe24418cde852439ce74c3dd2513c"
+				}
+			]
+		},
+		{
+			"name": "mpc",
+			"config-opts": [
+				"--enable-shared",
+				"--disable-static"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://ftp.gnu.org/gnu/mpc/mpc-1.0.3.tar.gz",
+					"sha256": "617decc6ea09889fb08ede330917a00b16809b8db88c29c31bfbb49cbf88ecc3"
+				}
+			]
+		},
+		{
+			"name": "gcc-fortran",
+			"config-opts": ["--enable-languages=fortran", "--disable-multilib"],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "http://mirrors.kernel.org/gnu/gcc/gcc-7.1.0/gcc-7.1.0.tar.gz",
+					"sha256": "3c187ef42e4cf657eec0ed641aac7f7e46261bcb1cea31b72dcf2dedddbbdea5"
+				}
+			]
+		},
+		{
+			"name": "gfortran-fix",
+			"buildsystem": "simple",
+			"sources": [
+				{
+					"type": "shell",
+					"commands": ["echo 'Symlinking libgfortran.so.4'"]
+				}
+			],
+			"build-commands": ["ln -s /app/lib64/libgfortran.so.4 /app/lib/libgfortran.so.4", "ln -s /app/lib64/libquadmath.so.0 /app/lib/libquadmath.so.0"]
+		},
+		{
+			"name": "pcre",
+			"config-opts": [
+				"--enable-shared"
+			],
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://ftp.pcre.org/pub/pcre/pcre-8.40.tar.gz",
```

Comment:
Without the module, the check for pcre_compile fails:

`configure:30592: checking for pcre_compile
configure:30592: gcc -o conftest -O2 -pthread -fopenmp  -L/app/lib  conftest.c --should-not-have-used-/usr/bin/pcre-config -lm   >&5
gcc: error: unrecognized command line option '--should-not-have-used-/usr/bin/pcre-config'
configure:30592: $? = 1
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "GNU Octave"
| #define PACKAGE_TARNAME "octave"
| #define PACKAGE_VERSION "4.2.1"
| #define PACKAGE_STRING "GNU Octave 4.2.1"
| #define PACKAGE_BUGREPORT "http://octave.org/bugs.html"
| #define PACKAGE_URL "http://www.gnu.org/software/octave/"
| #define PACKAGE "octave"
| #define VERSION "4.2.1"
| #define OCTAVE_SOURCE 1
| #define STDC_HEADERS 1
| #define HAVE_SYS_TYPES_H 1
| #define HAVE_SYS_STAT_H 1
| #define HAVE_STDLIB_H 1
| #define HAVE_STRING_H 1
| #define HAVE_MEMORY_H 1
| #define HAVE_STRINGS_H 1
| #define HAVE_INTTYPES_H 1
| #define HAVE_STDINT_H 1
| #define HAVE_UNISTD_H 1
| #define __EXTENSIONS__ 1
| #define _ALL_SOURCE 1
| #define _DARWIN_C_SOURCE 1
| #define _GNU_SOURCE 1
| #define _POSIX_PTHREAD_SEMANTICS 1
| #define __STDC_WANT_IEC_60559_ATTRIBS_EXT__ 1
| #define __STDC_WANT_IEC_60559_BFP_EXT__ 1
| #define __STDC_WANT_IEC_60559_DFP_EXT__ 1
| #define __STDC_WANT_IEC_60559_FUNCS_EXT__ 1
| #define __STDC_WANT_IEC_60559_TYPES_EXT__ 1
| #define __STDC_WANT_LIB_EXT2__ 1
| #define __STDC_WANT_MATH_SPEC_FUNCS__ 1
| #define _TANDEM_SOURCE 1
| #define YYTEXT_POINTER 1
| #define SEPCHAR ':'
| #define SEPCHAR_STR ":"
| #define SHELL_PATH "/bin/sh"
| #define OCTAVE_ENABLE_ATOMIC_REFCOUNT 1
| #define OCTAVE_IDX_TYPE int
| #define __NO_MATH_INLINES 1
| #define HAVE_FSEEKO 1
| #define _DARWIN_USE_64_BIT_INODE 1
| #define HAVE_PTHREAD_PRIO_INHERIT 1
| #define HAVE_PTHREAD 1
| #define HAVE_OMP_H 1
| #define HAVE_OMP_GET_NUM_THREADS 1
| #define OCTAVE_ENABLE_OPENMP 1
| #define HAVE_DLFCN_H 1
| #define LT_OBJDIR ".libs/"
| #define HAVE_LIBM 1
| #define F77_FUNC(name,NAME) name ## _
| #define F77_FUNC_(name,NAME) name ## _
| #define F77_USES_GFORTRAN_CALLING_CONVENTION 1
| #define HAVE_DLOPEN_API 1
| #define ENABLE_DYNAMIC_LINKING 1
| #define OCTAVE_ENABLE_FLOAT_TRUNCATE 1
| #define HAVE_PCRE_H 1
| /* end confdefs.h.  */
| /* Define pcre_compile to an innocuous variant, in case <limits.h> declares pcre_compile.
|    For example, HP-UX 11i <limits.h> declares gettimeofday.  */
| #define pcre_compile innocuous_pcre_compile
| 
| /* System header to define __stub macros and hopefully few prototypes,
|     which can conflict with char pcre_compile (); below.
|     Prefer <limits.h> to <assert.h> if __STDC__ is defined, since
|     <limits.h> exists even on freestanding compilers.  */
| 
| #ifdef __STDC__
| # include <limits.h>
| #else
| # include <assert.h>
| #endif
| 
| #undef pcre_compile
| 
| /* Override any GCC internal prototype to avoid an error.
|    Use char because int might match the return type of a GCC
|    builtin and then its argument prototype would still apply.  */
| #ifdef __cplusplus
| extern "C"
| #endif
| char pcre_compile ();
| /* The GNU C library defines this for functions which it implements
|     to always fail with ENOSYS.  Some functions are actually named
|     something starting with __ and the normal name is an alias.  */
| #if defined __stub_pcre_compile || defined __stub___pcre_compile
| choke me
| #endif
| 
| int
| main ()
| {
| return pcre_compile ();
|   ;
|   return 0;
| }
configure:30592: result: no
configure:30600: error: to build Octave, you must have the PCRE library and header files installed`